src/mainwindow.cpp

```diff
@@ -279,10 +278,18 @@ void MainWindow::closeEvent(QCloseEvent* event)
     QMainWindow::closeEvent(event);
 }
 
+const QString MainWindow::openDataFileName()
```

Comment:
const the method, not the return type

and I think it would be better to rename this function to `queryOpenDataFile`

Reply:
renaming sounds good - how do you want the signature to be - and how is this function different to other utility functions in this source file?

Reply:
```
QString MainWindow::queryOpenDataFile() const
```

I don't quite get what you mean by that question? are you saying there are utility functions that can be const which are not marked as const? fix that then please